partiql-parser/src/partiql.pest

```diff
@@ -2,18 +2,102 @@ WHITESPACE = _{ " " | "\t" | "\x0B" | "\x0C" | "\r" | "\n" }
 
 COMMENT = _{ LineComment | BlockComment }
 
-BlockComment = { "/*" ~ (BlockComment | (!"*/" ~ ANY))* ~ "*/" }
-LineComment = { "--" ~ (!"\n" ~ ANY )* ~ "\n" }
+BlockComment    = _{ "/*" ~ (BlockComment | (!"*/" ~ ANY))* ~ "*/" }
+LineComment     = _{ "--" ~ (!"\n" ~ ANY )* ~ "\n" }
 
 // TODO implement a full grammar, this is a very primitive version to start
 //      working with Pest and its APIs.
 
-// Entry point for full query parsing
-Query = _{ SOI ~ Token+ ~ EOI}
+//
+// Entry point for full query parsing.
+//
+Query = _{ SOI ~ Expression ~ EOI}
+
+//
+// Expressions
+//
+
+Expression = _{
+      SelectExpression
+    | ValueExpression
+}
+
+// TODO finish the SELECT statement
+
+SelectExpression = {
+    Select ~ (From ~ (Where)?)?
+}
+
+Select = _{
+    SelectValue
+}
+
+SelectValue = {
+    Select_ ~ Value_ ~ ValueExpression
+}
+
+// TODO joins
+
+From = {
+    From_ ~ ValueExpression ~ (As_)? ~ (Identifier)?
+}
+
+Where = {
+    Where_ ~ ValueExpression
+}
+
+// TODO all the operators / paths
+
+ValueExpression = {
+      Term
+    | LeftParen_ ~ Expression ~ RightParen_
+}
 
-// Entry point for query "scanning"
+// TODO fill in all the atoms
+
+Term = _{
+      NullLiteral
+    | BooleanLiteral
+    | String
+    | Number
+    | Identifier
+    | ArrayConstructor
+    | BagConstructor
+    | TupleConstructor
+}
+
+ArrayConstructor = {
+    LeftBracket_ ~ Values? ~ RightBracket_
+}
+
+BagConstructor = {
+    LeftDoubleAngle_ ~ Values? ~ RightDoubleAngle_
+}
+
+Values = _{
+    (ValueExpression ~ (Comma_ ~ ValueExpression)* ~ Comma_?)
```

Comment:
This would seem to support trailing commas?  Not sure that's intentional.

Reply:
It is--does this contradict what we did in the Kotlin grammar?

I do like this and I think we support it, but to be honest, I wasn't sure if we actually implemented it.

Reply:
Went and looked it up, we don't support it, I can remove this for now.

https://github.com/partiql/partiql-lang-kotlin/blob/7ac588979c68f33155b3df0fd37f01babc65f84f/lang/src/org/partiql/lang/syntax/SqlParser.kt#L2802-L2832